src/aiida/transports/transport.py

```diff
@@ -9,18 +9,22 @@
 """Transport interface."""
 
 import abc
+import asyncio
 import fnmatch
 import os
 import re
 import sys
 from collections import OrderedDict
-from pathlib import Path
+from pathlib import Path, PurePosixPath
+from typing import Optional, Union
 
 from aiida.common.exceptions import InternalError
 from aiida.common.lang import classproperty
 from aiida.common.warnings import warn_deprecation
 
-__all__ = ('Transport',)
+__all__ = ('AsyncTransport', 'Transport', 'TransportPath')
+
+TransportPath = Union[str, Path, PurePosixPath]
```

Comment:
To deal with generic path typing, it is better to cover more I think:

```python
PathLike = Union[AnyStr, os.PathLike]
```

In side the function, I'd rather all use `pathlib.Path` instead of `str`. The reason is we are all move to `pathlib.Path` in other module among the code base.